server/lib/nl/common/bad_words.py

```diff
@@ -12,40 +12,138 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import logging
-from typing import Set
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Dict, List
 
 from server.lib.config import GLOBAL_CONFIG_BUCKET
 from shared.lib import gcs
 
 BAD_WORDS_FILE = 'nl_bad_words.txt'
+_DELIM = ':'
+
+
+@dataclass
+class Entry:
+  # Set to true if the key of the Entry is a single word.
+  is_singleton: bool
+
+  # This entry has phrases beginning with the key of Entry.
+  # NOTE: These are full phrases including the word in the key,
+  # for convenience of matching.
+  #
+  # In case of "very bad dog", "very bad cow"
+  # We will have: ["very bad dog", "very bad cow"]
+  #
+  phrases: List[str] = field(default_factory=list)
+
+  # This is a multi-word entry, where we must match every
+  # word (in any order) in the query.  When sorted, the first
+  # word is the key of Entry, and the remaining words are here.
+  #
+  # In case of "idiot:cat:bat", only if idiot, bat and cat
+  # appear in the query (regardless of ordering),
+  # will we match.  We will have: ["cat", "idiot"]
+  other_words: List[str] = field(default_factory=list)
+
+
+@dataclass
+class BadWords:
```

Comment:
as a nit, since this is a new class, perhaps a good time to call this "BannedWords"? That's essentially what they are and feels more "standard"?

Reply:
Good idea.  Changed class name, but keeping function names the same to match file name.

Reply:
Thanks!